sky/packages/sky/lib/src/animation/animation_performance.dart

```diff
@@ -31,12 +50,22 @@ enum AnimationStatus {
 /// may also take direct control of the timeline by manipulating [progress], or
 /// [fling] the timeline causing a physics-based simulation to take over the
 /// progression.
-class AnimationPerformance {
+class AnimationPerformance implements WatchableAnimationPerformance {
   AnimationPerformance({ AnimatedVariable variable, this.duration }) :
     _variable = variable {
     _timeline = new Timeline(_tick);
   }
 
+  /// Returns an [AnimationPerformanceView] that wraps this performance,
+  /// so that a pointer to this object can be passed around without
+  /// allowing users of that pointer to mutate the AnimationPerformance state.
+  AnimationPerformanceView _view;
+  AnimationPerformanceView get view {
```

Comment:
I've done this, but Ivan pointed out that actually this doesn't get us as much safety as possible, because Dart's dynamic typing means that only the analyzer will catch errors. Even in checked mode, code will just run if it tries to call AnimationPerformance methods on a WatchableAnimationPerformance variable.